packages/js/src/executors/tsc/tsc.impl.ts

```diff
@@ -87,7 +92,11 @@ export async function* tscExecutor(
     const disposePackageJsonChanged = await watchForSingleFileChanges(
       join(context.root, projectRoot),
       'package.json',
-      () => updatePackageJson(options, context, target, dependencies)
+      () => {
+        if (packageJsonFileExists) {
```

Comment:
I'm pretty sure this check is not needed as `watchForSingleFileChanges` doesn't actually read the file, it just checks if the paths are matching a changed file:
https://github.com/nrwl/nx/blob/master/packages/js/src/utils/watch-for-single-file-changes.ts#L11-L21

Reply:
Yeah, this ones not needed. Can you remove it? This wouldn't be called if the packageJson didn't exist.

Reply:
Done